dashboard/client/src/SideNavBar.svelte

```diff
@@ -113,61 +44,46 @@
   {#if visible}
     <div transition:slide={{ axis: 'x' }} class="sidebar pt-5 flex-shrink-0">
       <button type="button" class="btn-close close-btn" on:click={() => (visible = false)}></button>
-      {#if challenges === undefined || classes === undefined}
+      {#if !($challenges.length && $classes.length)}
```

Comment:
In here I think it's better to check for the undefined value. Happened already couple times that some challenges/classes broke the DB initialisation and so later the API then returned empty set of classes and challenges. In this case the UI would keep showing "loading" eventhough the loading was done, right? But of course that ideally should not happen

Reply:
yes, that is right - I'm changing the initial value to an explicit `null` and then checking for coercion to false rather than explicit equality with `undefined`, though